### DISCUSSION TITLE
ALE only works when i save the file with :w

### DISCUSSION THREAD

**Discussion by sorrowlol** [w0rp/ale#4121] [Q&A - Ask for help with problems]
Is there a way for this to be automatic without having to save the file? I installed it using Pathogen.

Thank you!

**Comment by sorrowlol** [comment]
To be specific: I am using it with Solidity which is listed as supported. I Tried it with C++ and it works as expected.

**Comment by hsanson** [comment]
Checkout these help documents:

- :h ale-lint
- :h g:ale_lint_on_save
- :h ale-lint-file-linters

Not sure but seems solium tool requires files to be on disk for linting, therefore the only way for ALE to lint using it is after saving the file.

For solc and solhint it seems it should work while typing if ALE is properly configured.
